### PR TITLE
chat system messages for rentetion policy changes

### PR DESCRIPTION
**Pull request by joshblum**
posts a new `Changerentention` `SYSTEM` message to the conversation/channel/#general when a retention policy is changed. cli renders these messages, UI to follow
in action:
impconv:
```
$ kb chat retention-policy test13 --explode 1d
Found PRIVATE CHAT conversation: test1,test13
Set the conversation retention policy?
Hit Enter to confirm, or Ctrl-C to cancel.
Found PRIVATE CHAT conversation: test1,test13
Messages will explode after 1 day by default
$ kb chat read test13
Found PRIVATE CHAT conversation: test1,test13
[1]  [test1 6s]   [test1 changed the conversation retention policy. Messages will explode after 1 day by default]

$ kb chat retention-policy test13 --keep
Found PRIVATE CHAT conversation: test1,test13
Set the conversation retention policy?
Hit Enter to confirm, or Ctrl-C to cancel.
Found PRIVATE CHAT conversation: test1,test13
Messages will be retained indefinitely
$ kb chat read test13
Found PRIVATE CHAT conversation: test1,test13
[1]  [test1 38s]   [test1 changed the conversation retention policy. Messages will explode after 1 day by default]                                                                                                                                                                                                                                                        
[2]   [test1 6s]   [test1 changed the conversation retention policy. Messages will be retained indefinitely]                                                                                                                                                                                                                                                              

$ kb chat retention-policy test13 --expire 1w
Found PRIVATE CHAT conversation: test1,test13
Set the conversation retention policy?
Hit Enter to confirm, or Ctrl-C to cancel.
Found PRIVATE CHAT conversation: test1,test13
Messages will expire after 7 days
$ kb chat read test13
Found PRIVATE CHAT conversation: test1,test13
[1]  [test1 54s]   [test1 changed the conversation retention policy. Messages will explode after 1 day by default]                                                                                                                                                                                                                                                        
[2]  [test1 22s]   [test1 changed the conversation retention policy. Messages will be retained indefinitely]                                                                                                                                                                                                                                                              
[3]   [test1 7s]   [test1 changed the conversation retention policy. Messages will expire after 7 days]

```

team:
```
$ kb chat retention-policy new --explode 5m
Set the team-wide retention policy?
Hit Enter to confirm, or Ctrl-C to cancel.
Team policy: Messages will explode after 5 minutes by default
#general channel: Uses the team policy
$ kb chat read new
[1]  [test1 2m]   [test1 changed the team retention policy. Messages will explode after 5 minutes by default]

$ kb chat retention-policy new --channel new-chan --inherit
Set the channel retention policy?
Hit Enter to confirm, or Ctrl-C to cancel.
Team policy: Messages will explode after 5 minutes by default
#new-chan channel: Uses the team policy
$ kb chat read new --channel new-chan
[1]  [test1 11s]   [test1 changed the channel retention policy to inherit from the team policy. Messages will explode after 5 minutes by default]

$ kb chat retention-policy new --channel new-chan --expire 1m
Set the channel retention policy?
Hit Enter to confirm, or Ctrl-C to cancel.
Team policy: Messages will explode after 5 minutes by default
#new-chan channel: Messages will expire after 30 days
$ kb chat read new --channel new-chan
[1]  [test1 2m]   [test1 changed the channel retention policy to inherit from the team policy. Messages will explode after 5 minutes by default]                                                                                                                                                                                                                          
[2]  [test1 2m]   [test1 changed the channel retention policy. Messages will expire after 30 days]  
```

### CODE REVIEW

File: go/chat/server.go [hunk 1]
```diff
@@ -2078,21 +2078,104 @@ func (h *Server) AddTeamMemberAfterReset(ctx context.Context,
...
+	if err != nil {
+		return err
+	}
+	switch typ {
+	case chat1.RetentionPolicyType_INHERIT:
+		isInherit = true
+	}
+
+	p := chat1.Pagination{Num: 1}
+	ib, err := h.G().InboxSource.ReadUnverified(ctx, uid, types.InboxSourceDataSourceAll, &chat1.GetInboxQuery{
```
Comment (mmaxim) [inline go/chat/server.go]: Can use `GetUnverifiedConv`

File: go/chat/server.go [hunk 2]
```diff
@@ -2078,21 +2078,104 @@ func (h *Server) AddTeamMemberAfterReset(ctx context.Context,
...
+	if isInherit {
+		teamRetention := conv.TeamRetention
+		if teamRetention == nil {
+			policy = chat1.RetentionPolicy{}
+		} else {
+			policy = *teamRetention
+		}
+	}
+	membersType := conv.Metadata.MembersType
+	info, err := CreateNameInfoSource(ctx, h.G(), membersType).LookupName(
```
Comment (mmaxim) [inline go/chat/server.go]: Maybe this RPC should just take a name instead of ID

Comment (joshblum) [reply go/chat/server.go]: we'd still have to do a lookup to send the convID to gregor so i don't think it buys us much

File: go/chat/server.go (hunk 1 above)
Comment (joshblum) [reply go/chat/server.go]: done

File: shared/constants/chat2/message.js [hunk 3]
```diff
@@ -550,6 +550,14 @@ const uiMessageToSystemMessage = (minimum, body, reactions): ?Types.Message => {
         ...minimum,
       })
     }
+    case RPCChatTypes.localMessageSystemType.changeretention: {
```
Comment (joshblum) [inline shared/constants/chat2/message.js]: will have a full version of this in the next PR, but flow was unhappy with the new type missing from the switch

**Comment by joshblum** [comment]
kbfs server failed to build, merging

**Pull request merged**